From dd566bbfaf40644809feefb96f58f160e99a78b9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ville=20Syrj=C3=A4l=C3=A4?= <ville.syrjala@linux.intel.com>
Date: Thu, 19 Jan 2023 20:32:22 +0200
Subject: [PATCH 087/105] modesetting: Don't feed stack garbage to the kernel
 in LUT reserved fields
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Stop putting stack garbage into the gamma LUT blob reserved
fields.

Fixes: 245b9db03a1e ("modesetting: Use GAMMA_LUT when available")
Signed-off-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
---
 hw/xfree86/drivers/modesetting/drmmode_display.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/xfree86/drivers/modesetting/drmmode_display.c b/hw/xfree86/drivers/modesetting/drmmode_display.c
index 48f8e5f5e..cfe77079e 100644
--- a/hw/xfree86/drivers/modesetting/drmmode_display.c
+++ b/hw/xfree86/drivers/modesetting/drmmode_display.c
@@ -1922,6 +1922,7 @@ drmmode_set_gamma_lut(drmmode_crtc_private_ptr drmmode_crtc,
         lut[i].red = red[i];
         lut[i].green = green[i];
         lut[i].blue = blue[i];
+        lut[i].reserved = 0;
     }
 
     if (drmModeCreatePropertyBlob(drmmode->fd, lut, sizeof(lut), &blob_id))
-- 
2.45.2

